Feat(#61): 아이콘 크기 및 더보기 버튼 간격 조정

hyeopsang opened the pull request:
## 📌 이슈 넘버
#61 

## 📄 작업 페이지 및 내용 요약
아이콘 크기를 변경하고 더보기 버튼의 글자와 아이콘 사이 간격을 조정합니다.
## 📝 작업 내용
- 아이콘 size-4 => size-5
- 더보기 버튼 gap-0.5 => gap-1

## 📸 스크린샷(선택)
<img width="98" height="29" alt="image" src="https://github.com/user-attachments/assets/e4ad40a5-19ba-4ffe-8765-4b24809bdd66" />

hyeopsang committed:
Refactor(#61): 폴더 구조 변경

hyeopsang committed:
feat(#61): 파일 경로 수정

hyeopsang committed:
Feat(#61): 댓글 작성 및 무한 스크롤 구현

hyeopsang committed:
Feat(#61): 댓글 및 답글 API 통합 및 React Query 최적화
- 댓글/답글의 CRUD 로직을 한 곳에서 관리하도록 통합
- invalidateQueries를 활용하여 데이터 변경 시 자동 리프레시 로직 적용
- 답글 생성시 부모 댓글의 ID를 기준으로 특정 쿼리 키만 무효화하도록 최적화
- 더보기 버튼 방식의 답글 조회 로직 구현

hyeopsang committed:
Refactor(#61): 댓글, 대댓글 관련 api 로직 분리

hyeopsang committed:
Refactor(#61): 폴더 구조 변경

hyeopsang committed:
Feat(#61): 댓글 수정 기능 추가
- 폴더 구조 리팩터링
- 댓글 수정 기능 구현
- 수정 모드 상태 관리용 Context 생성

hyeopsang committed:
Refactor(#61): 댓글 관련 context 통합 및 컴포넌트 네이밍 변경

hyeopsang committed:
Feat(#61): ErrorFallback 컴포넌트 생성 및 댓글, 답글 에러 처리 추가

hyeopsang committed:
Refactor(#61): 댓글 에러 처리 및 상태 관리 개선
- CommentInteractionProvider 적용 범위 수정
- 댓글 관련 mutation 에러 핸들링 추가
- 상대 시간 계산 시 미래 시점 보정 처리
- 댓글 수정 시 변경사항 없을 경우 취소 처리
- useCommentAction isPending 반환 및 disabled 상태 적용

hyeopsang committed:
Feat(#61): 댓글 관리 로직 작업
- 댓글 접기, 댓글 펼치기 버튼 추가
- 댓글 더보기 ui 변경
- 프로필 기본 값 사진 변경
- chevron down svg 추가

hyeopsang committed:
Refactor(#61): 댓글 쿼리키 훅 분리 및 관련 context 제거

hyeopsang committed:
Merge remote-tracking branch 'origin/develop' into feat/#61-comment

hyeopsang committed:
Refactor(#61): 댓글 상태 관리 zustand 전환 및 BaseCommentItem 공용화
- 기존 훅 기반 댓글 상태 관리 로직을 zustand store로 리팩토링
- 댓글/답글에서 공통으로 사용하는 UI 로직을 BaseCommentItem으로 분리

hyeopsang committed:
Refactor(#61): 댓글, 답글 zustand store 통합
- useEffect를 사용해 props와 state를 동기화하는 패턴 제거
- BaseCommentItem에서 관리하던 isEditing 상태를 CommentEditForm 내부로 이동
- useCommentEdit, useCommentReply 파일 삭제

hyeopsang committed:
Refactor(#61): 댓글/답글 렌더링 구조 개선
- 답글 보기 클릭 시 답글 리스트 렌더링
- BaseCommentItem UI 구조 리팩토링
- 답글 영역이 열린 경우에만 답글 수정 가능하도록 제한

hyeopsang committed:
Refactor(#61): useInfiniteScroll => react-virtuoso로 변경

hyeopsang committed:
Refactor(#61): CommentForm css 변경

hyeopsang committed:
Refactor(#61): 댓글 시간 표기를 상대 시간(formatRelativeTime)으로 통일

hyeopsang committed:
Feat(#61): 댓글 관련 스토리북 테스트 코드 작성
-  storybook/test 설치
- 댓글 스켈레톤 ui 컴포넌트 추가
- 댓글, 댓글 스켈레톤 스토리북 테스트 로직 작성

hyeopsang committed:
Refactor(#61): 댓글 유저 권한 관리를 위한 Context 도입 및 Prop Drilling 제거
- CommentUserContex 및 Hook 생성 (로그인 유저 보 및 본인 확인 로직 중앙화)
- CommentContainer Provider 적용
- BaseCommentItem 등 하위 컴포넌트에서 불필 한 props 제거
- Context 도입에 따른 댓글 스토리북 로직 수정

hyeopsang committed:
Refactor(#61): 폴더 구조 변경

hyeopsang committed:
Merge branch 'develop' into feat/#61-comment

Kim-HyounSuk committed:
Merge remote-tracking branch 'origin/develop' into feat/#61-comment

Kim-HyounSuk committed:
Merge remote-tracking branch 'origin/develop' into feat/#61-comment

Kim-HyounSuk committed:
Chore(#61): 아이콘 크기 고정 제거(reload, chevron-down)

Kim-HyounSuk committed:
Chore(#61): 아이콘 에셋 크기 지정

hyeopsang committed:
chore(#61): 아이콘 크기 및 간격 조정

Kim-HyounSuk commented on src/providers/CurrentUserContext.tsx:
상협님! 코드 잘 봤습니다.
한 가지 제안드리고 싶은 게 있는데요.
이 부분은 `SessionProvider`와 역할이 겹치는 것 같아서, Provider 대신 Custom Hook으로 분리하는 게 더 나을 것 같습니다.
불필요한 렌더링도 줄이고 구조도 더 단순해질 것 같은데 어떻게 생각하시나요?

hyeopsang committed:
Refactor(#61): CurrentUserContext 제거하고 useCurrntUser 훅 생성

hyeopsang committed:
Refactor(#61): 사용하지 않는 파일 제거 및 폴더 이동
- useInfiniteScroll 훅 제거
- CommentHeader 컴포넌트 제거
- BaseCommentItem 스토리북 및 스켈레톤 컴포넌트 폴더 이동 (CommentItem 폴더 => BaseCommentItem)
